release-content/0.14/release-notes/10756_hooks_and_observers.md

```diff
@@ -0,0 +1,59 @@
+<!-- Hooks: https://github.com/bevyengine/bevy/pull/10756 -->
+<!-- Observers: https://github.com/bevyengine/bevy/pull/10839 -->
+
+As much as we love churning through homogeneous blocks of data in a tight loop here at Bevy; not every task is a perfect fit for the straightforward ECS model.
+Responding to changes and/or processing events are vital tasks in any application, and games are no exception.
+
+Bevy has accumulated a number of subtly distinct tools to handle this over the years, and in 0.14 is picking up two more.
+Let's quickly go over the existing tools to contextualize why adding more is useful:
+
+- Buffered [`Event`]s:multiple-producer, multiple-consumer queues. Flexible and efficient, but requires polling. Events are dropped after two frames.
+- Change detection via [`Added`] and [`Changed`]: Granular, automatic, great integration with queries. Linear time checks, no added data.
+- [`RemovedComponents`]: A special form of event that is triggered when a component is removed from an entity, or an entity with that component is despawned.
+
+All of these (and systems themselves!) use a ["pull"-style mechanism]: metaphorical events are sent regardless of whether or not anyone is listening, and listeners must periodically poll to ask if anything has changed.
+This is a useful pattern, and one we intend to keep around!
+By polling, we can process events in batch, getting more context and improving data locality (which makes the CPU go brr).
+
+But it comes with some limitations:
+
+- polling has a small but non-zero overhead every frame
+- there is an unavoidable delay between an event being triggered and the response being processed
+
+This delay is the critical problem:
+
+- data (like indexes or hierarchies) exists, even momentarily, in an invalid state
+- we can't process arbitrary chains of recursive logic within a single cycle, no matter how carefully we order our event writers and readers
+
+To overcome these limitations, Bevy 0.14 introduces two complementary "push"-style mechanisms inspired by the ever-wonderful C ECS, [flecs]:
+
+1. [**Component Lifecycle Hooks:**](https://dev-docs.bevyengine.org/bevy/ecs/component/struct.ComponentHooks.html) Mutations to the world that automatically occur in response to **lifecycle events**: whenever a component of the given type is added, overwritten or removed.
+2. [**Observers:**](https://dev-docs.bevyengine.org/bevy/ecs/observer/struct.Observer.html) On-demand systems that listen to [`Trigger`] events, which commonly target specific entities. These systems are run sequentially and recursively at the next command flush point in response to a trigger being sent.
+
+Only one hook per component type can be registered, and it cannot be overwritten.
+It is intended for use by library authors, to strictly enforce invariants (such as hierarchy correctness) required to use that type.
```

Comment:
```suggestion
These are intended for enforcing lower level ECS invariants required to use components (Eg. hierarchy correctness). They're comparable to constructors and destrcutors. These will typically be used by library authors.
```
There's some useful patterns with hooks that doesn't require one to be a library author so idk how much it should be stressed who will typically be using them more. One thing I'm concerned about here is the mention of them being used for hierarchy invariants... hierarchy doesn't use them yet. The work to make hierarchy use hooks is straight forward but is just soft blocked [as I mentioned here](https://github.com/bevyengine/bevy/issues/12235#issuecomment-2116031592). It's just people reading this will be looking for examples & we'd leave them lost if we mentioned a hierarchy use that doesn't exist.